clang/lib/Driver/Driver.cpp

```diff
@@ -790,41 +793,60 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
             continue;
           }
 
-          // Store the current triple so that we can check for duplicates in the
-          // following iterations.
-          FoundNormalizedTriples[NormalizedName] = Val;
+          // Store the current triple so that we can check for duplicates in
+          // the following iterations.
+          FoundNormalizedTriples[NormalizedName] = NormalizedName;
+          UniqueSYCLTriplesVec.push_back(TT);
+        }
+      } else
+        Diag(clang::diag::warn_drv_empty_joined_argument)
+            << SYCLTargetsValues->getAsString(C.getInputArgs());
+    }
+    // -fsycl-add-targets is a list of paired items (Triple and file) which are
+    // gathered and used to be linked into the final device binary. This can
+    // be used with -fsycl-targets to put together the final conglomerate binary
+    if (SYCLAddTargets) {
+      if (SYCLAddTargets->getNumValues()) {
+        // Use of -fsycl-add-targets adds additional files to the SYCL device
+        // link step.  Regular offload processing occurs below
+        for (StringRef Val : SYCLAddTargets->getValues()) {
+          // Parse out the Triple and Input (triple:binary) and create a
+          // ToolChain for each entry.
+          // The expected format is 'triple:file', any other format will
+          // not be accepted.
+          std::pair<StringRef, StringRef> I = Val.split(':');
+          if (!I.first.empty() && !I.second.empty()) {
+            llvm::Triple TT(I.first);
+            if (TT.getArch() == llvm::Triple::UnknownArch || !TT.isSPIR()) {
+              Diag(clang::diag::err_drv_invalid_sycl_target) << Val;
```

Comment:
```suggestion
              Diag(clang::diag::err_drv_invalid_sycl_target) << I.first;
```
This should improve the diagnostic.